File: frontend/src/types/stats.ts
import type { DeployStatus } from './deploy'

export interface RecentDeployItem {
  id: string
  router_hostname: string | null
  router_ip: string | null
  status: DeployStatus
  deployed_at: string
}

export interface DashboardStats {
  routers_total: number
  routers_active: number
  templates_total: number
  deploys_total: number
  deploys_last_30d: number
  deploys_success_last_30d: number
  deploys_failed_last_30d: number
  recent_deploys: RecentDeployItem[]
}

File: backend/app/schemas/stats.py
import uuid
from datetime import datetime
from pydantic import BaseModel
from app.models.config_history import DeployStatus


class RecentDeployItem(BaseModel):
    id: uuid.UUID
    router_hostname: str | None
    router_ip: str | None
    status: DeployStatus
    deployed_at: datetime


class DashboardStats(BaseModel):
    routers_total: int
    routers_active: int
    templates_total: int
    deploys_total: int
    deploys_last_30d: int
    deploys_success_last_30d: int
    deploys_failed_last_30d: int
    recent_deploys: list[RecentDeployItem]
